ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/velocity_set_path.cpp

```diff
@@ -172,24 +145,33 @@ void VelocitySetPath::changeWaypointsForStopping(int stop_waypoint, int obstacle
 
     // v = (v0)^2 + 2ax, and v0 = obstacle velocity
     double changed_vel = std::sqrt(obstacle_velocity * obstacle_velocity + 2.0 * deceleration * calcInterval(index, stop_waypoint));
-
     double prev_vel = prev_waypoints_.waypoints[index].twist.twist.linear.x;
-    if (changed_vel > prev_vel)
-    {
-      new_waypoints_.waypoints[index].twist.twist.linear.x = prev_vel;
-    }
-    else
-    {
-      new_waypoints_.waypoints[index].twist.twist.linear.x = changed_vel;
-    }
+    new_waypoints_.waypoints[index].twist.twist.linear.x = std::min(prev_vel, changed_vel);
   }
 
   // fill velocity with obstacle velocity for stopping
   for (int i = stop_waypoint; i <= obstacle_waypoint; i++)
   {
     new_waypoints_.waypoints[i].twist.twist.linear.x = new_waypoints_.waypoints[stop_waypoint].twist.twist.linear.x;
   }
+}
+
+void VelocitySetPath::changeWaypointsForDeceleration(double deceleration, int closest_waypoint, int obstacle_waypoint)
+{
+  double square_vel_min = decelerate_vel_min_ * decelerate_vel_min_;
+  int extra = 4; // for safety
```

Comment:
why is this extra = 4 just here in the code? and not part of distance for stopline or distance for obstacle?

Reply:
this code block was just moved in this file and not part of them.

Reply:
I saw that it was just moved but my question is why was this originally in? Another hack?

Does not do the same as:
- stop_distance_obstacle
- stop_distance_stopline
?

Reply:
just hack, and no relationship with “stop..”